browser/brave_content_browser_client.cc

```diff
@@ -175,8 +177,11 @@ using extensions::ChromeContentBrowserClientExtensionsPart;
 #endif
 
 #if !defined(OS_ANDROID)
+#include "brave/browser/brave_ads/brave_ads_host.h"
```

Comment:
there's no need for two header files or two different classes, you just need two different implementations

Reply:
I created Android implementation and it has slightly different header with android specific fields:
https://github.com/brave/brave-core/pull/9983/files#diff-668263f48bdfad831d830b5e34154c416ff711549a1d10d6efadc2e9ef2fdacb

brave_ads_host.h header also has desktop specific fields. 
Maybe we should have two headers?

Reply:
well, looking around this is done in different ways throughout the codebase. In this case I think it's ok to have two different headers/classes, but the guards need to be fixed as mentioned for gn. Should be OS_ANDROID for brave_ads_host_android and ENABLE_EXTENSIONS for brave_ads_host.h